tests/test_zuora_discovery.py

```diff
@@ -0,0 +1,129 @@
+
+        for stream in streams_to_test:
+            with self.subTest(stream=stream):
+                catalog = next(iter([catalog for catalog in found_catalogs
+                                     if catalog["stream_name"] == stream]))
+                assert catalog  # based on previous tests this should always be found
+                # gather expectations
+                expected_replication_keys = self.expected_replication_keys()[stream]
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_replication_method = self.expected_replication_method()[stream]
+                expected_automatic_fields = expected_primary_keys | expected_replication_keys
+                
+                if stream in self.additional_automatic_field_in_streams:
+                    expected_automatic_fields.add('TransactionDate')
+
+                # gather results
+                schema_and_metadata = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])
+                metadata = schema_and_metadata["metadata"]
+                schema = schema_and_metadata["annotated-schema"]
+                stream_properties = [item for item in metadata if item.get("breadcrumb") == []]
+                actual_replication_keys = set(stream_properties[0].get(
+                    "metadata", {self.REPLICATION_KEYS: []}).get(self.REPLICATION_KEYS, [])
+                )
+                actual_primary_keys = set(stream_properties[0].get(
+                    "metadata", {self.PRIMARY_KEYS: []}).get(self.PRIMARY_KEYS, [])
+                )
+                actual_replication_method = stream_properties[0].get(
+                    "metadata", {self.REPLICATION_METHOD: None}).get(self.REPLICATION_METHOD)
+
+
+                # verify there is only 1 top level breadcrumb
+                self.assertTrue(len(stream_properties) == 1,
+                                msg="There is NOT only one top level breadcrumb for {}".format(stream) + \
+                                "\nstream_properties | {}".format(stream_properties))
+
+                actual_fields = []
+                for md_entry in metadata:
+                    if md_entry['breadcrumb'] != []:
+                        actual_fields.append(md_entry['breadcrumb'][1])
+
+                # Verify there are no duplicate/conflicting metadata entries.
+                self.assertEqual(len(actual_fields), len(set(actual_fields)), msg = "duplicates in the metadata entries retrieved")
+
+                # verify replication key(s)
+                self.assertSetEqual(expected_replication_keys, actual_replication_keys)
+
+                # verify primary key(s)
+                self.assertSetEqual(expected_primary_keys, actual_primary_keys)
+
+                # verify the actual replication matches our expected replication method
+                self.assertEqual(expected_replication_method, actual_replication_method) # BUG_TDL-9711
```

Comment:
`# BUG_TDL-9711` looks unrelated, can be removed.

Reply:
removed

Reply:
It is still there on L#104 and L#106